dockstore-client/src/main/java/io/dockstore/client/cli/nested/CromwellLauncher.java

```diff
@@ -81,24 +88,46 @@ public void initialize() {
 
     @Override
     public List<String> buildRunCommand() {
-        final List<String> cwlSpecificFlags = Arrays.asList("--type", "cwl");
-        final List<String> runCommand;
-        runCommand = Lists.newArrayList(primaryDescriptor.getAbsolutePath(), "--inputs", provisionedParameterFile.getAbsolutePath());
-        // NOTE: Support for ZIP imports exists, but we decided to comment it out for now as it was causing some issues.
-        //runCommand = Lists.newArrayList(primaryDescriptor.getAbsolutePath(), "--inputs", provisionedParameterFile.getAbsolutePath(), "--imports", zippedEntry.getAbsolutePath());
-        final String[] s = { "java", "-jar", executionFile.getAbsolutePath(), "run" };
         List<String> arguments = new ArrayList<>();
-        arguments.addAll(Arrays.asList(s));
-        arguments.addAll(runCommand);
+        arguments.add("java");
+
+        if (cromwellVmOptions.size() > 0) {
+            List<String> cromwellVmOptionsParsed = cromwellVmOptions.stream().map(string -> string.split(",")).flatMap(Arrays::stream).map(this::trimAndPrintInput)
+                    .collect(Collectors.toList());
+            arguments.addAll(cromwellVmOptionsParsed);
+        }
+
+        // Cromwell help specifies the 'run' command line format as: run [options] workflow-source
+        Collections.addAll(arguments, "-jar", executionFile.getAbsolutePath(), "run", "--inputs", provisionedParameterFile.getAbsolutePath());
+
+        // NOTE: Support for ZIP imports exists, but we decided to comment it out for now as it was causing some issues.
+        //Collections.addAll(arguments, "--imports", zippedEntry.getAbsolutePath());
+
         // There are currently issues with Cromwell 44 automatic type detection.  If it's CWL, the flag must be added.
         // https://github.com/broadinstitute/cromwell/issues/5085
         // Remove these 3 lines once the type can be detected again.
         if (languageType == DescriptorLanguage.CWL) {
-            arguments.addAll(cwlSpecificFlags);
+            Collections.addAll(arguments, "--type", "cwl");
+        }
+
+        if (cromwellExtraParameters.size() > 0) {
+            List<String> cromwellExtraParametersParsed = cromwellExtraParameters.stream().map(string -> string.split(",")).flatMap(Arrays::stream).map(this::trimAndPrintInput)
+                    .collect(Collectors.toList());
+            arguments.addAll(cromwellExtraParametersParsed);
         }
+
+        // Add workflow source file
+        arguments.add(primaryDescriptor.getAbsolutePath());
+
         return arguments;
     }
 
+    private String trimAndPrintInput(String input) {
+        String trimmedInput = input.trim();
+        System.out.println(trimmedInput);
```

Comment:
1. This feels like a side effect to me; I would probably not have the println, and then you could just get rid of this method and have `String::trim` in your streams code.
2. Regardless of whether you decide to do that or not, it looks like the output in the console is going to be a list of extra arguments and nothing else, but maybe I'm reading it wrong. Could you attach a screenshot of what the output looks like?

Reply:
Removed the method and replaced it with String::trim